### PR TITLE
Update clam version

### PR DESCRIPTION
**Pull request by steverendell**
When this PR is merged, it will push new :latest images. Then Ill be able to remove the temporary docker pushes with another PR

### CODE REVIEW

File: .github/workflows/docker-build-push.yml [hunk 1]
```diff
@@ -71,6 +71,15 @@ jobs:
           labels: ${{ steps.meta.outputs.labels }}
           push: true
 
+      - name: Push container image to temporary tag
+        if: ${{ success() && github.ref != 'refs/heads/main' }}
+        uses: docker/build-push-action@v2
+        with:
+          context: ./${{ matrix.dockerfile-dir }}
+          tags: ${{ steps.meta.outputs.tags }}-test
```
Comment (steverendell) [inline .github/workflows/docker-build-push.yml:79]: temporary section needed to use as base image for freshclam

File: .github/workflows/clamav-daily.yml [hunk 2]
```diff
@@ -72,6 +72,16 @@ jobs:
           labels: ${{ steps.meta.outputs.labels }}
           push: true
 
+      - name: Push container image to repository
+        if: ${{ success() && github.ref != 'refs/heads/main' }}
+        uses: docker/build-push-action@v2
+        with:
+          file: ./${{ matrix.dockerfile-dir }}/Dockerfile.freshclammed
+          context: ./${{ matrix.dockerfile-dir }}
+          tags: ${{ steps.meta.outputs.tags }}-test
```
Comment (steverendell) [inline .github/workflows/clamav-daily.yml:81]: temporary section needed to push image for local testing

File: hmpps-clamav/Dockerfile [hunk 3]
```diff
@@ -9,17 +10,36 @@ COPY kubernetes.repo /etc/yum.repos.d/kubernetes.repo
...
+    wget -nv --user-agent="${USER_AGENT}" https://www.clamav.net/downloads/production/clamav-${CLAM_VERSION}.tar.gz.sig &&  \
+    wget -nv https://cmake.org/files/v3.22/cmake-3.22.0.tar.gz && \
     gpg --import /tmp/talos.pub && \
     gpg --decrypt clamav-${CLAM_VERSION}.tar.gz.sig && \
-    tar xzf clamav-${CLAM_VERSION}.tar.gz
+    tar xzf clamav-${CLAM_VERSION}.tar.gz && \
+    tar xzf cmake-3.22.0.tar.gz
+
+RUN cd cmake-3.22.0 && \
+    ./bootstrap && make && make install
```
Comment (steverendell) [inline hmpps-clamav/Dockerfile:35]: clamav now requires cmake to build it

File: hmpps-clamav/Dockerfile [hunk 4]
```diff
@@ -47,7 +67,7 @@ RUN mkdir /var/run/clamav && \
 
 USER 1000
 
-RUN freshclam --no-dns && \
+RUN freshclam --help && freshclam --no-dns && \
```
Comment (steverendell) [inline hmpps-clamav/Dockerfile:70]: will remove with pr taking out the temp pushes

File: hmpps-clamav/Dockerfile.freshclammed [hunk 5]
```diff
@@ -1,4 +1,4 @@
-FROM ghcr.io/ministryofjustice/hmpps-clamav:latest
+FROM ghcr.io/ministryofjustice/hmpps-clamav:latest-test
```
Comment (steverendell) [inline hmpps-clamav/Dockerfile.freshclammed:1]: trivy now succeeds

File: .github/workflows/clamav-daily.yml (hunk 2 above)
Comment (steverendell) [reply .github/workflows/clamav-daily.yml:81]: Image tested using pathfinder integration tests ✔️

File: hmpps-clamav/Dockerfile [hunk 6]
```diff
@@ -1,6 +1,7 @@
 FROM centos:latest
 
-ENV CLAM_VERSION=0.103.2
+ENV CLAM_VERSION=0.104.1
+ENV USER_AGENT="Mozilla/5.0 (Windows NT 10.0; Win64; x64) Gecko/20100101 Chrome/96.0.4664.45"
```
Comment (petergphillips) [inline hmpps-clamav/Dockerfile:4]: cheeky :grin:

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**